src/sgp/SGP.cc

```diff
@@ -629,6 +631,7 @@ void TerminationHandler()
 			SLOGE("Game has been terminated due to an unknown error");
 		}
 	}
+	shutdownGame();
```

Comment:
Just curious - what happens if it is not explicitly shut down? I thought the OS would automatically clean up everything

Reply:
I the case of any other system than Android: Yes. In the case of Android: As the game is used as a library there, the OS does not clean up anything. So that means that the SoundCallback thread and FillBuffer thread continue to run and thus the nice JA2 music continues to play 😉 .

But right now I missed some cases where it is not cleaned up yet (because the `TerminationHandler` returns early).

Reply:
But in general I think it makes sense to try shut down in this case, because it e.g. also cleans the temp dir and things like that. It just leaves everything in a cleaner state...

Reply:
And memory checkers don't complain about still bound memory on exit.

Reply:
True. I refactored the `TerminationHandler` a little, now it should clean up in all situations...